### DISCUSSION TITLE
how to handle dynamic properties

### DISCUSSION THREAD

**Discussion by balazssoltesz** [MichalLytek/type-graphql#879] [Q&A]
Hello, 

I try to figure out, how should i handle dynamic properties in an Object/InputType. My main problem is i declared languages in my app's config file, and want to generate ObjectType and InputType based on my used languages in the app. 
So i have an array with available languages: 

`const languages =  ["en", "de"] as const;`

An I wanna generate types based on this array:

```
@ObjectType()
class IntlObject {
  @Field()
  [lang: typeof languages[number]]: string;
}
```

But it doesn't work. Any idea, how can i handle this issue?

**Comment by MichalLytek** [comment]
You should describe the languages as an enum and then create an object type with `lang: Language` and `value: string` (or label, whatever works for you).

So then you can in runtime return proper data, like:
```ts
return {
  data: "...",
  intl: {
    lang: Language.EN,
    value: "foo bar",
  },
};
```

**Reply by balazssoltesz** [reply]
Thanks for your answer! 
I'm using typegoose with mongoose-intl plugin, where localizable fields described like 

```
title: {
  en: 'foo',
  de: 'bar',
}
```

so i have to convert data format to db and back for every localizable field?

**Reply by MichalLytek** [reply]
Or you can create the `IntlObject` with normal `en` and `de` properties, without using dynamic enum.

**Reply by plcardonab** [reply]
@balazssoltesz  are you implementing server internationalization ? could you provide me some guidance of how to implement it ?
